### PR TITLE
added reply routes

### PR DESCRIPTION
**Pull request by clarkohw**

### CODE REVIEW

**Comment by clarkohw** [comment]
looks like my auto formatter is not a fan of ' vs " but besides that most of the changes are in replies.js

File: backend/routes/replies.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+/**
+ * get all replies that user has responded to.
+ * Body should include userId.
+ */
+router.get("/", body("userId").isUUID(), async (req, res) => {
+  Reply.findAll({
+    where: { "$User.id$": req.body.userId },
+    include: [
+      { model: User, attributes: [] },
+      { model: Request, attributes: ["id", "name"] },
```
Comment (NicholasSpringer) [inline backend/routes/replies.js:22]: We should not include userId in requests bodies, instead refer to `req.user.id`, as the `user` field of `req` is filled in by the express jwt middleware, meaning the user will not be able to modify the ID in its issued jwt token.

File: backend/routes/replies.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    //check if the user exists, if so create reply and add to user.
+
+    try {
+      reply = await Reply.create({
+        id: replyId,
+      });
+
+      let user = await User.findOne({ where: { id: req.body.userId } });
+      if (user == null) {
+        return res.status(StatusCodes.NOT_FOUND).send("User not found.");
```
Comment (NicholasSpringer) [inline backend/routes/replies.js:60]: Same thing here

File: backend/routes/replies.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(StatusCodes.NOT_FOUND).json({ errors: errors.array() });
+    }
+
+    let reply;
+    try {
+      reply = await Reply.findOne({
+        where: { id: req.body.replyId, "$User.id$": req.body.userId },
```
Comment (NicholasSpringer) [inline backend/routes/replies.js:109]: And same thing here

File: backend/routes/requests.js [hunk 4]
```diff
@@ -1,211 +1,210 @@
...
+  try {
+    reqResource = await Request.findOne({
+      where: { id: req.params.requestId },
+      include: { model: User, attributes: ["id"] },
+    });
+
+    if (reqResource.User.id != req.user.id) {
+      // Authenticated user does not own the request resource
+      return res.sendStatus(StatusCodes.FORBIDDEN);
     }
```
Comment (NicholasSpringer) [inline backend/routes/requests.js:200]: Here we should also standardize on 4 space indentation

File: backend/routes/users.js [hunk 5]
```diff
@@ -1,104 +1,110 @@
...
+    console.log("devlog");
+    // Create token that expires in 24 hours
+    let token = jwt.sign({ id: req.body.userId }, JWT_SECRET, {
+      expiresIn: 86400,
+    });
+    return res.status(StatusCodes.OK).json({
+      token: token,
+    });
+  }
 );
```
Comment (NicholasSpringer) [inline backend/routes/users.js:108]: Same here

File: backend/config.env [hunk 6]
```diff
@@ -0,0 +1,6 @@
+DB_NAME = "lonliness"
+DB_USERNAME= postgres
+DB_HOST = localhost
+DB_PASSWORD = 
+PORT = 5000
+JWT_SECRET = test
```
Comment (NicholasSpringer) [inline backend/config.env:6]: Most of this information is dependent on the developer or should be kept secret for production environments, so I don't think we should commit .env files to the repository. Perhaps add *.env to the gitignore

File: backend/package-lock.json [hunk 7]
```diff
@@ -0,0 +1,1205 @@
+{
```
Comment (NicholasSpringer) [inline backend/package-lock.json:1]: I think we should stick to only using yarn

**Review by NicholasSpringer — COMMENTED** [review]
Looks mostly good, but I think a few changes are needed. 
1. For formatting, I think we should standardize on 4 spaces, since it seems like that is what Trevor and Ivo are using as well. 
2. I noticed in the replies routes the express jwt middleware functionality isn't actually getting used.
3. I don't think we should commit the .env config. Also, I couldn't find where you are actually importing the config using dotenv, so might want to add that.
4. A `package-lock` file was added, I believe that is from npm, but if we are using yarn already we should remove that.

**Pull request merged**